.github/workflows/test-e2e.yaml

```diff
@@ -0,0 +1,76 @@
+name: E2E Tests
+
+on:
+  workflow_call:
+
+jobs:
+  build-docker-image:
+    name: Build Docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and export
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          tags: vechain/thor:${{ github.sha }}
+          outputs: type=docker,dest=/tmp/vechain-thor.tar
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: vechain-thor-image-${{ github.sha }}
+          path: /tmp/vechain-thor.tar
+          retention-days: 7
+
+  run-tests:
+    runs-on: ubuntu-latest
+    needs: build-docker-image
+    env:
+      THOR_IMAGE: vechain/thor:${{ github.sha }}
+    name: Run Tests
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: vechain/thor-e2e-tests
+          # https://github.com/vechain/thor-e2e-tests/tree/2cb22d804bb3cdf075917dbece42a182d42d7486
+          ref: 2cb22d8
```

Comment:
so instead of targeting main we can target certain commit for tests (or even branch), is there anyway we can automate it instead of using hardcoded value ? something like specifying it before running workflow

Reply:
Hhmm, how do you mean? Where would we specify it?

Reply:
thats a good question, which I don't have answer too :)